src/services/assetManager.ts

```
@@ -1,104 +1,58 @@
-      const assetPath = path.join(this.javaAgentDir, asset.name);
-      if (await fileExists(assetPath)) return;
-
-      const lockfilePath = await this.createLockfile(asset.name);
-      if (!lockfilePath) {
-        this.log(`Did not download ${asset.name} because lockfile already exists.`);
-        return;
-      }
-
-      this.log(`Downloading AppMap Java agent ${asset.name}`);
-      await GithubRelease.downloadAsset(asset, assetPath);
-
-      const symlinkPath = path.join(this.javaAgentDir, 'appmap.jar');
-      try {
-        unlinkSync(symlinkPath);
-      } catch (e) {
-        // if the symlink that we're trying to remove does not exist, don't do anything
-      }
-
-      const symlinkCreated = this.updateLatestFile(asset.name, assetPath, symlinkPath);
-      unlinkSync(lockfilePath);
+  symlinked(assetName: string, symlinkPath: string): void {
+    this.log(`Symlinked ${assetName} to ${symlinkPath}`);
+  }
 
-      this.log(`Finished downloading AppMap Java agent to ${assetPath}`);
-      Telemetry.sendEvent(DOWNLOADED_JAVA_JAR, { symlinkCreated, version: asset.name });
-    } catch (e) {
-      const err = e as Error;
-      this.log(err.message);
+  error(err: Error): void {
+    this.log(err.message);
 
-      Telemetry.sendEvent(DEBUG_EXCEPTION, {
-        exception: err,
-        errorCode: ErrorCode.AssetAcquisitionFailure,
-        log: this.logLines.join('\n'),
-      });
-    }
+    Telemetry.sendEvent(DEBUG_EXCEPTION, {
+      exception: err,
+      errorCode: ErrorCode.AssetAcquisitionFailure,
+      log: this.logLines.join('\n'),
+    });
   }
 
-  public static async register(): Promise<void> {
-    await this.getLatestJavaJar();
+  protected log(line: string) {
+    this.logLines.push(line);
+    if (this.outputChannel) this.outputChannel.appendLine(line);
   }
 }
+
+export default async function installLatestJavaJar(raiseOnError: boolean) {
```

When run in the background, errors are swallowed and reported via telemetry. When run as a user command, the exception is handled by the caller.